### PR TITLE
Weight loss program

### PR DESCRIPTION
**Pull request by samanpwbb**
This is ready to go. Decided against drastic conceptual changes.

// mb-pages
CSS: 262 kB (minified) => 29.7 kB (gzipped)
SVG: 50 kB => 15.1 kB (gzipped)

// This branch
CSS: 172 kB (minified) => 21.9 kB (gzipped)
SVG: 50 kB => 15.1 kB (gzipped)

---

This branch turned into a larger v1 push, beyond just space saving. Changes:

- Adds missing --dark modifiers to some typography rules.
- Stroked buttons and select elements now have a 1px stroke by default.
- Adds new `btn--stroked--2` and `select--stroked--2` modifier classes to create stroked buttons and select elements that have 2px strokes.
- [internal only] Renames `neutral` variables to more meaningful `disabled-{type}-interactive` variables.
- Darken the `gray-faint` color to #e5e5e5 so it's legible on a wider range of monitors.
- Removes teal color variations.
- Removes 600px layout scale value for padding and margin.
- Refactors range inputs to use currentColor for coloring, which save space
- Removes `darken5` and `lighten5` color variations to save space and encourage more accessible design. 
- Removes `p{n}` and `m{n}` rules.
- Limits the possible colors available for form elements for accessibility reasons and to save size. `-faint` or `-light` form elements are no longer available for _any element_, and `darken10` and `darken10` are no longer available for any for element except buttons.
- Adds distinct hover states for active checkboxes and radio inputs.
- Redesigns checkbox according to #745.

### CODE REVIEW

**Comment by samanpwbb** [comment]
@tristen @davidtheclark this is ready for review.

File: site/catalog/toggles.js [hunk 1]
```diff
@@ -84,12 +78,12 @@ class Toggles extends React.Component {
         </div>
 
         <div className='mb12 txt-bold color-darken50 txt-uppercase txt-s'>Light variations</div>
-        <div className='bg-gray round p12y mt12'>
+        <div className='bg-gray round px12 py12y mt12'>
```
Comment (tristen) [inline site/catalog/toggles.js]: Should be `py12`

File: site/catalog/toggles.js [hunk 2]
```diff
@@ -84,12 +78,12 @@ class Toggles extends React.Component {
...
         {lightenColors.map((c) => <ToggleEl key={c} color={c} />)}
         </div>
 
         <div className='mt24 mb12 txt-bold color-darken50 txt-uppercase txt-s'>Disabled and light</div>
-        <div className='bg-gray round p12y mt12'>
+        <div className='bg-gray round px12 py12y mt12'>
```
Comment (tristen) [inline site/catalog/toggles.js]: Should be `py12`

File: site/examples/progress_bars.js [hunk 3]
```diff
@@ -9,7 +9,7 @@ const playback = `<div class='flex-parent flex-parent--row flex-parent--center-c
     <button class='flex-child py18 pr12 pl18 border-r border--gray-light bg-white bg-green-faint-on-hover cursor-pointer round-l'>
       <div class='triangle--r triangle color-green color-green'></div>
     </button>
-    <div class='flex-child flex-child--grow p18 bg-white'>
+    <div class='flex-child flex-child--grow px18 py18bg-white'>
```
Comment (tristen) [inline site/examples/progress_bars.js]: Missing space between class names

File: site/examples/tooltips.js [hunk 4]
```diff
@@ -2,22 +2,22 @@ import React from 'react';
...
 const basic = `<div class='flex-parent-inline flex-parent--center-cross flex-parent--column'>
-  <div class='flex-child p6 bg-darken75 color-white align-center  round txt-bold txt-s'>Hello world!</div>
+  <div class='flex-child px6 py6 bg-darken75 color-white align-center  round txt-bold txt-s'>Hello world!</div>
   <span class='flex-child triangle triangle--d'></span>
 </div>`;
 
 const basic_right = `<div class='flex-parent-inline flex-parent--center-cross'>
   <span class='flex-child triangle triangle--l'></span>
-  <div class='flex-child p6 round txt-bold txt-s align-center bg-darken75 color-white'>Hello world!</div>
+  <div class='flex-child px6 py6round txt-bold txt-s align-center bg-darken75 color-white'>Hello world!</div>
```
Comment (tristen) [inline site/examples/tooltips.js]: Missing space between class names

File: site/examples/tooltips.js [hunk 5]
```diff
@@ -2,22 +2,22 @@ import React from 'react';
...
 </div>`;
 
 const basic_left = `<div class='flex-parent-inline flex-parent--center-cross'>
-  <div class='flex-child p6 round txt-bold txt-s align-center bg-darken75 color-white'>Hello world!</div>
+  <div class='flex-child px6 py6round txt-bold txt-s align-center bg-darken75 color-white'>Hello world!</div>
```
Comment (tristen) [inline site/examples/tooltips.js]: Missing space between class names

File: site/examples/tooltips.js [hunk 6]
```diff
@@ -2,22 +2,22 @@ import React from 'react';
...
   <span class='flex-child triangle triangle--r'></span>
 </div>`;
 
 const basic_with_closure = `<div class='flex-parent-inline flex-parent--center-cross flex-parent--column'>
-  <div class='flex-child relative p18 round txt-bold txt-s align-center bg-darken75 color-white'>
+  <div class='flex-child relative px18 py18round txt-bold txt-s align-center bg-darken75 color-white'>
```
Comment (tristen) [inline site/examples/tooltips.js]: Missing space between class names

File: src/typography.css [hunk 7]
```diff
@@ -39,13 +38,10 @@ textarea {
   font-weight: normal;
 }
 
+.txt-kbd--dark,
 .prose--dark kbd {
-  background: var(--lighten5);
-  border: 1px solid var(--lighten10);
-  line-height: 18px;
-  border-radius: 3px;
-  padding: 2px 3px;
```
Comment (tristen) [inline src/typography.css:47]: Was dropping these properties intentional?

File: src/variables.json [hunk 8]
```diff
@@ -60,9 +53,9 @@
   "white": "#fff",
   "transparent": "transparent",
 
-  "neutral10": "rgba(127, 127, 127, 0.1)",
-  "neutral25": "rgba(127, 127, 127, 0.25)",
-  "neutral45": "rgba(127, 127, 127, 0.45)",
+  "disabled-primary-interactive-color": "rgba(127, 127, 127, 0.25)",
+  "disabled-secondary-interactive-color": "rgba(127, 127, 127, 0.1)",
+  "disabled-text-color": "rgba(127, 127, 127, 0.45)",
```
Comment (tristen) [inline src/variables.json]: Is `disabled` accurate? This is used for placeholder too right? And I'm not sure how `primary` & `secondary` align with how they are used. How about `input-{value}-on-blur`?

File: site/catalog/toggles.js [hunk 9]
```diff
@@ -2,27 +2,21 @@ import React from 'react';
 
 const colors = [
   null,
-  'gray-faint',
-  'gray-light',
   'gray',
-  'gray-dark',
```
Comment (tristen) [inline site/catalog/toggles.js:8]: Curious about all these `gray-{varient}` omissions? They ... still exist but you're removing them from the catalog?

**Review by tristen — CHANGES_REQUESTED** [review]
Changes look great!

> Decided against drastic conceptual changes

By that do you mean reducing color variations on form elements to two states?

File: site/catalog/toggles.js (hunk 9 above)
Comment (samanpwbb) [reply site/catalog/toggles.js:8]: This was a fix - we never had -dark variants here.

File: src/typography.css (hunk 7 above)
Comment (samanpwbb) [reply src/typography.css:47]: Yeah, prose--dark is a modifier for prose so no need to duplicate these rules.

File: src/variables.json (hunk 8 above)
Comment (samanpwbb) [reply src/variables.json]: Ohh that is the one inaccurate use. I can change this to be 'placeholder-text-color'

**Comment by samanpwbb** [comment]
Addressed feedback!

File: scripts/build-color-variants.js [hunk 10]
```diff
@@ -52,13 +47,11 @@ const allColors = [
   'purple-light',
   'purple-faint',
 
-  'darken5',
   'darken10',
   'darken25',
   'darken50',
   'darken75',
 
-  'lighten5',
```
Comment (davidtheclark) [inline scripts/build-color-variants.js:61]: I like these cuts!

File: scripts/build-color-variants.js [hunk 11]
```diff
@@ -73,8 +66,15 @@ function isSemitransparent(color) {
...
+function isNotSuitableForForms(color) {
+  // Do not create form elements from values that are too light or too dark,
+  // for accessibility and to save space.
+  return color === 'black' || /^(darken10|lighten10)$/.test(color) || /(-dark|-light|-faint)$/.test(color);
+}
+
+function isNotSuitableForButtons(color) {
+  // Filled Buttons should be allowed to have subtle backgrounds within reason.
+  return color === 'black' || /(-faint|-dark)$/.test(color);
 }
```
Comment (davidtheclark) [inline scripts/build-color-variants.js]: 🎉

File: scripts/build-color-variants.js [hunk 12]
```diff
@@ -300,51 +281,59 @@ function buildColorVariants(variables, config) {
...
+        }
+
+        input:checked + .switch--dot-${color}::after {
+          background-color: var(--${color});
+        }
       `);
     }, '');
   };
 
   variantGenerators.range = function (colors) {
```
Comment (davidtheclark) [inline scripts/build-color-variants.js:328]: Dark and non-dark ranges seem to look the same:

screen shot 2017-06-04 at 8 22 01 am

File: scripts/build-color-variants.js [hunk 13]
```diff
@@ -73,8 +66,15 @@ function isSemitransparent(color) {
   return /^(lighten|darken)/.test(color);
 }
 
-function isDark(color) {
-  return color === 'black' || /-dark$/.test(color);
+function isNotSuitableForForms(color) {
+  // Do not create form elements from values that are too light or too dark,
+  // for accessibility and to save space.
+  return color === 'black' || /^(darken10|lighten10)$/.test(color) || /(-dark|-light|-faint)$/.test(color);
+}
```
Comment (davidtheclark) [inline scripts/build-color-variants.js]: What do you about think `-faint` for hover and active states? Should we keep those, or also remove them to encourage stronger contrast?

**Comment by davidtheclark** [comment]
Since a lot changes here, might be useful to add a Changelog modifications in this PR.

File: scripts/build-color-variants.js (hunk 12 above)
Comment (samanpwbb) [reply scripts/build-color-variants.js:328]: Good catch - was from an earlier refactor attempt. I removed this class but didn't remove the documentation?

File: scripts/build-color-variants.js (hunk 13 above)
Comment (samanpwbb) [reply scripts/build-color-variants.js]: I'd be in favor of cutting them - can add that to this PR

File: scripts/build-color-variants.js (hunk 12 above)
Comment (samanpwbb) [reply scripts/build-color-variants.js:328]: Done!

**Comment by samanpwbb** [comment]
@tristen @davidtheclark oookay i think this is ready for a final 👀  then merge

**Review by davidtheclark — APPROVED** [review]
All these changes seem wonderful to me! Very fine work.

**Pull request merged**